src/command.rs

```diff
@@ -126,22 +133,45 @@ impl ShellCommand for BuiltinCommand {
         self.tokens[1..].iter().map(|s| s.resolve()).collect()
     }
 
-    fn pipe(&self, stdin: Option<ChildStdout>) -> Result<Option<ChildStdout>, Box<dyn Error>> {
+    fn pipe(
+        &self,
+        stdin: Option<ChildStdout>,
+        aliases: &mut HashMap<String, String>,
+    ) -> Result<Option<ChildStdout>, Box<dyn Error>> {
         let (pipe_out_r, pipe_out_w) = pipe()?;
         let (pipe_err_r, pipe_err_w) = pipe()?;
+        let (alias_r, alias_w) = pipe()?;
 
         match unsafe { fork() }? {
             ForkResult::Parent { child: _ } => {
                 drop(pipe_out_w);
                 drop(pipe_err_w);
+                drop(alias_w);
+
+                // Read updated aliases from child
+                let mut alias_reader = unsafe { File::from_raw_fd(alias_r.into_raw_fd()) };
+                let mut serialized_aliases = String::new();
+                alias_reader.read_to_string(&mut serialized_aliases)?;
+                if !serialized_aliases.is_empty() {
+                    *aliases = serde_json::from_str(&serialized_aliases)?;
+                }
+
                 Ok(Some(ChildStdout::from(pipe_out_r)))
             }
             ForkResult::Child => {
                 drop(pipe_out_r);
                 drop(pipe_err_r);
+                drop(alias_r);
+
                 dup2(pipe_out_w.as_raw_fd(), 1)?;
                 dup2(pipe_err_w.as_raw_fd(), 2)?;
-                self.run_builtin(stdin)?;
+                self.run_builtin(stdin, aliases)?;
+
+                // Send updated aliases to parent
+                let serialized_aliases = serde_json::to_string(aliases)?;
+                let mut alias_writer = unsafe { File::from_raw_fd(alias_w.into_raw_fd()) };
+                write!(alias_writer, "{}", serialized_aliases)?;
+
```

Comment:
This is really cool haha. I'm assuming that this is for this particular scenario:

```bash
> alias r='echo' | r Hello
Hello
```

But this doesn't work in Bash, so we don't necessarily need to support it? Might cut down on the complexity.